### ISSUE TITLE
Not Recognizing VSCode as Microsoft Signed PopOS 19.04

### ISSUE THREAD

**Issue by rtablada** [OmniSharp/omnisharp-vscode#3021]
## Issue Description ##

After installing a fresh install of PopOS 19.04 (based on Ubuntu 19.04) and installing VSCode + Ominsharp I am getting an error that the debugger can not run in this version of VSCode. Then the debugging process stops. This is the case whether running via "Debug Test" or using the launch.json settings for C# projects.

Originally I installed VSCode from the PopShop then updated via the official MS download page for VSCode.
The license in both the .deb and in the "About VSCode" seem to be the proper Microsoft signed licence.

I have also tried uninstalling VSCode, using apt-get purge on VSCode and I have removed the following directories:

```
~/.vscode
~/.config/Code
~/.omnisharp
```

Then I installed again from the fresh .deb file on the official Microsoft VS Code page.

If there is some other signing or config that is still showing this as a different distribution of VSCode I'm not able to figure out why/where that may be.

## Steps to Reproduce ##

* Install Ubuntu 19.04
* Install VSCode
* Install dotnet sdk and mono-complete
* Install C# extension
* Try to debug test or app

## Expected Behavior ##

Debugger should work as usual and start debugging.

## Actual Behavior ##

Debugger prints

```
-------------------------------------------------------------------
You may only use the Microsoft .NET Core Debugger (vsdbg) with
Visual Studio Code, Visual Studio or Visual Studio for Mac software
to help you develop and test your applications.
-------------------------------------------------------------------
```

### OmniSharp log ###
<details>Post the output from Output-->OmniSharp log here</details>

### C# log ###
<details>Post the output from Output-->C# here</details>

## Environment information ##

**VSCode version**: 1.33.1
**C# Extension**: 1.18.0

<details><summary>Mono Information</summary>
        OmniSharp using global mono :5.20.1</details>
<details><summary>Dotnet Information</summary>
.NET Core SDK (reflecting any global.json):
 Version:   2.2.203
 Commit:    e5bab63eca

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  19.04
 OS Platform: Linux
 RID:         linux-x64
 Base Path:   /home/ryan/dotnet/sdk/2.2.203/

Host (useful for support):
  Version: 2.2.4
  Commit:  f95848e524

.NET Core SDKs installed:
  2.2.203 [/home/ryan/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.2.4 [/home/ryan/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.2.4 [/home/ryan/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.2.4 [/home/ryan/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|
|---|---|---|
|atom-keybindings|ms-vscode|3.0.6|
|babelrc|waderyan|1.0.0|
|bracket-pair-colorizer|CoenraadS|1.0.61|
|change-case|wmaurer|1.0.0|
|code-beautifier|michelemelluso|2.2.1|
|code-settings-sync|Shan|3.2.9|
|code-spell-checker|streetsidesoftware|1.7.5|
|color-highlight|naumovs|2.3.0|
|csharp|ms-vscode|1.18.0|
|csharpextensions|jchannon|1.3.0|
|EditorConfig|EditorConfig|0.13.0|
|ember-syntax|dhedgecock|0.5.0|
|githistory|donjayamanne|0.4.6|
|gitignore|codezombiech|0.6.0|
|Handlebars|andrejunges|0.4.1|
|markdown-all-in-one|yzhang|2.2.0|
|markdown-table-formatter|josa|0.2.3|
|Material-theme|zhuangtongfa|2.21.0|
|one-monokai|azemoh|0.3.7|
|prettier-vscode|esbenp|1.8.1|
|rainbow-csv|mechatroner|1.0.0|
|rainglow|daylerees|1.5.2|
|ruby|rebornix|0.22.3|
|sass-lint|glen-84|1.0.5|
|seti-icons|qinjia|0.1.3|
|theme-cobalt2|wesbos|2.1.6|
|Theme-MarkdownKit|ms-vscode|0.1.4|
|url-encode|flesler|1.1.0|
|vscode-circleci|jvandyke|0.5.0|
|vscode-ember-unstable|lifeart|0.2.25|
|vscode-eslint|dbaeumer|1.8.2|
|vscode-fileutils|sleistner|2.13.7|
|vscode-github|KnisterPeter|0.30.2|
|vscode-new-file|dkundel|4.0.2|
|vscode-sql-beautify|sensourceinc|0.0.4|
|vscode-table-formatter|shuworks|1.2.1|
|vscode-text-pastry|jkjustjoshing|1.2.0|
|vscode-toggle-quotes|BriteSnow|0.3.2|;

</details>

**Comment by rchande** [comment]
cc @gregg-miskelly

**Comment by gregg-miskelly** [comment]
@rtablada Are you getting an error message? Or is the debugger just exiting?

If the debugger is just exiting - I bet you are running into https://github.com/OmniSharp/omnisharp-vscode/issues/3010. There is a private version of the extension at the end of that issue which you can try to see if this fixes the problem for you.

If you are getting an error - can you grab a log? https://github.com/OmniSharp/omnisharp-vscode/wiki/Enabling-C%23-debugger-logging

**Comment by matfantinel** [comment]
I'd like to hop in and say that I'm having a similar issue on my Ubuntu 19.04. Both VS Code and .NET Core have been installed through their snap packages. (tried with .deb too, nothing changed).

Unfortunately I don't have access to my computer right now to give more detailed information, but besides the message:
```
-------------------------------------------------------------------
You may only use the Microsoft .NET Core Debugger (vsdbg) with
Visual Studio Code, Visual Studio or Visual Studio for Mac software
to help you develop and test your applications.
-------------------------------------------------------------------
```
I also get this one, related to Chromium: `InitializeSandbox() called with multiple threads in process gpu-process.`. My launch.json is set to start my API on a browser, but it seems to be an issue with Chromium used in VS Code, since my default browser is Firefox.

I thought it would have something to do with them being Snap packages, but the official versions don't run on a sandbox.

@gregg-miskelly @rtablada I am confident we have the same issue. Does that message help at all? I can try to get a more detailed log this evening.

**Comment by gregg-miskelly** [comment]
@matfantinel please [install the latest beta](https://github.com/OmniSharp/omnisharp-vscode/wiki/Installing-Beta-Releases). You are running into #3010.

**Comment by matfantinel** [comment]
@gregg-miskelly It worked. Thanks a lot!

**Comment by lonix1** [comment]
I also have this problem (just upgraded to ubuntu 19.04)

The latest release is `1.19.1` which doesn't work for me.
Did you get it working with that release?

**Comment by gregg-miskelly** [comment]
@lonix1 sorry this issue should have been closed earlier. I left it open in case the original opener wanted to comment more. The known issue with Ubuntu 19.04 was fixed in C# extension 1.19.1, so whatever issue you were running into is different. The problem you are running into is that the debugger is abruptly exiting? If so, I am happy to try and help troubleshoot. Please open a new issue, and include a debugger log ([instructions])https://github.com/OmniSharp/omnisharp-vscode/wiki/Enabling-C%23-debugger-logging)).
